ledger_device_sdk/src/nbgl.rs

```diff
@@ -279,8 +199,34 @@ impl<'a> NbglHomeAndSettings<'a> {
         Reply: From<<T as TryFrom<ApduHeader>>::Error>,
     {
         unsafe {
-            if NVM_REF.is_some() {
-                self.settings_contents = nbgl_content_t {
+            loop {
+                let info_contents: Vec<*const c_char> = self
+                    .info_contents
+                    .iter()
+                    .map(|s| s.as_ptr())
+                    .collect::<Vec<_>>();
+
+                let info_list: nbgl_contentInfoList_t = nbgl_contentInfoList_t {
+                    infoTypes: INFO_FIELDS.as_ptr() as *const *const c_char,
+                    infoContents: info_contents[1..].as_ptr() as *const *const c_char,
+                    nbInfos: INFO_FIELDS.len() as u8,
+                };
+
+                let icon: nbgl_icon_details_t = match self.glyph {
+                    Some(g) => g.into(),
+                    None => nbgl_icon_details_t::default(),
+                };
+
+                for (i, setting) in self.setting_contents.iter().enumerate() {
+                    SWITCH_ARRAY[i].text = setting[0].as_ptr();
+                    SWITCH_ARRAY[i].subText = setting[1].as_ptr();
+                    SWITCH_ARRAY[i].initState =
+                        NVM_REF.as_mut().unwrap().get_ref()[i] as nbgl_state_t;
```

Comment:
No check on NVM_REF as discussed yesterday ?

Reply:
Not needed: as soon as some setting are defined through the `settings` call, NVM_REF is set.
So if no settings are defined, the `for` loop will be empty and NVM_REF not used...